modules/contact/src/actions/ContactAction.C

```diff
@@ -1028,6 +1071,129 @@ ContactAction::addNodeFaceContact()
   }
 }
 
+void
+ContactAction::createSidesetPairsFromGeometry()
+{
+  mooseInfo("The contact action is reading the list of sidesets and matching them assuming that: "
+            "1) Those sidesets wrap around the surfaces of the bodies, 2) Axisymmetry of the "
+            "geometry and the mesh exists. Deviations from axisymmetry may give rise to wrong "
+            "contact pairing. The user is encouraged to check the screen output to confirm pairs "
+            "were assigned correctly.");
+
+  _mesh = &_problem->mesh();
+
+  if (!_mesh)
+    mooseError("Internal error in contact action.");
+
+  if (!_mesh->getMesh().is_serial())
+    paramError(
+        "sideset_list",
+        "The generation of automatic contact pairs in the contact action requires a serial mesh.");
+
+  // Compute centers of gravity for each sideset
+  std::vector<std::pair<BoundaryName, Point>> sideset_list_cog;
+  const auto & sideset_ids = _mesh->meshSidesetIds();
+  // _mesh->buildBndElemList();
+  const auto & bnd_to_elem_map = _mesh->getBoundariesToActiveSemiLocalElemIds();
+
+  for (const auto & sideset_name : _sideset_list)
+  {
+    // If the sideset provided in the input file isn't in the mesh, error out.
+    const auto find_set = sideset_ids.find(_mesh->getBoundaryID(sideset_name));
+    if (find_set == sideset_ids.end())
+      paramError("sideset_list",
+                 "The sideset ",
```

Comment:
It might actually be a boundary that is in the mesh, but is not a sideset, so this message should be adjusted:
"<sideset_name> is not defined as a sideset in the mesh"
I'd drop the part about "Please revise your input file." because that goes without saying for any param error.